app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/FileItem.java

```diff
@@ -0,0 +1,94 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.net.Uri;
+import android.os.Parcel;
+import android.os.Parcelable;
+import androidx.annotation.IntDef;
+
+import androidx.annotation.NonNull;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.FileItem.FileStatus.*;
+
+/**
+ * Helper class, part of the local file sharing module.
+ *
+ * Defines a file-item to represent the files being transferred.
+ */
+public class FileItem implements Parcelable {
+
+  @Retention(RetentionPolicy.SOURCE)
+  @IntDef({ TO_BE_SENT, SENDING, SENT, ERROR })
+  public @interface FileStatus {
+    int TO_BE_SENT = -1;  // File yet to be sent
+    int SENDING = 0;      // Being sent
+    int SENT = +1;        // Successfully sent
+    int ERROR = +2;       // Error encountered while transferring the file
+  }
+
+  private Uri fileUri;
+  private String fileName;
+  private int fileStatus;
+
+  public FileItem(@NonNull Uri fileUri) { // For sender devices
+    this.fileUri = fileUri;
+    this.fileName = WifiDirectManager.getFileName(fileUri);
+    this.fileStatus = TO_BE_SENT;
+  }
+
+  public FileItem(@NonNull String fileName) { // For receiver devices
+    this.fileUri = null;
+    this.fileName = fileName;
+    this.fileStatus = TO_BE_SENT;
+  }
+
+  // For making FileItem a Parcelable
+  @Override
+  public int describeContents() {
+    return 0;
+  }
+
+  @Override
+  public void writeToParcel(@NonNull Parcel dest, int flags) {
+    dest.writeString(fileName);
```

Comment:
you aren't putting uri in the parcel also why does this need to be parcelable?

Reply:
Thanks for pointing out, this is from when I was using fragments (the TransferProgressFragment had a ArrayList<FileItem> field to be obtained from the arguments bundle)

Should I update the code (since its just a couple more lines) or remove it altogether (since its not required to be Parcelable)?

Reply:
Remove it, parcelable impls are a thing of the past because of (you guessed it) 💯  KOTLIN 💯 
and the fantastic "Parcelize" annotation.
```@Parcelize
data class Language constructor(
  val id: Long = 0,
  var active: Boolean,
  var occurencesOfLanguage: Int,
  var language: String,
  var languageLocalized: String,
  var languageCode: String,
  var languageCodeISO2: String
) : Parcelable
```
a full class with getters/setters/toString/hashCode/equals and it is parcelable.

Reply:
A month ago I wasn't sure whether learning kotlin would be _that_ beneficial to me. Now I have some semblance of how much easier life will be with it
[Removed](https://github.com/kiwix/kiwix-android/pull/1276/commits/4b260e15ed27aa4f9329826ca90da5c4a81e51f3) the parcelable code